covsirphy/cleaning/jhu_complement.py

```diff
@@ -224,10 +271,9 @@ def _recovered_full(self, df):
             sel_2 = df[self.C].diff().diff().rolling(14).mean() > 0
             cf_diff = (df[self.C] - df[self.F]).rolling(14).mean()
             sel_3 = df[self.R] < 0.01 * cf_diff
-            sel_4 = df[self.R] > 0.99 * cf_diff
-            s_df_1 = df.loc[sel_C1 & sel_2 & sel_4]
-            s_df_2 = df.loc[sel_R1 & sel_2 & sel_3]
-            if s_df_1.empty and s_df_2.empty:
+            s_df = df.loc[sel_C1 & sel_R1 & sel_2 & sel_3]
+            recovery_period_is_valid = self._validate_recovery_period(df)
+            if s_df.empty and recovery_period_is_valid:
```

Comment:
Please replace line 375 and 276 with `if s_df.empty and self._validate_recovery_period(df):` to avoid calculation of `self._validate_recovery_period(df)` when `s_df.empty`.